torch/ao/quantization/pt2e/port_metadata_pass.py

```diff
@@ -98,6 +98,19 @@ def _port_metadata_for_input_quant_nodes(
         q_node, dq_node = _find_q_dq_node_for_user(input_node, node)
         if q_node is None or dq_node is None:
             return
+        # add metadata for all the node between q_node and get_attr node
+        # if the q_node can be traced back to get_attr node
+        q_to_get_attr_nodes = [q_node]
+        q_node_input = q_node.args[0]
+        while isinstance(q_node_input, torch.fx.Node) and q_node_input.op not in [
+            "placeholder",
+            "get_attr",
+        ]:
+            q_to_get_attr_nodes.append(q_node_input)
+            q_node_input = q_node_input.args[0]
+        if isinstance(q_node_input, torch.fx.Node) and q_node_input.op == "get_attr":
+            for n in q_to_get_attr_nodes:
+                _add_metadata(n, node)
```

Comment:
I had a comment here which I dont see now, but you dont want to add metadata from the 'node'. one you need to port metadata from is get_attr. So for a pattern like `get_attr -> q -> dq -> linear` the q node should get metadata from get_attr and not linear node. This is also why in the test we should test for having ported the right metadata, not just existence of metadata

Reply:
oh I see, OK will fix